openmetadata-service/src/main/java/org/openmetadata/service/search/elasticsearch/ElasticSearchClient.java

```diff
@@ -414,6 +419,193 @@ public Response search(SearchRequest request) throws IOException {
     }
   }
 
+  @Override
+  public Response searchWithHierarchy(SearchRequest request) throws IOException {
```

Comment:
I think this code is again very much the same , we can bifurcate the different things to methods , and see if we can resuse those